frontend/components/canvas.tsx

```diff
@@ -1,75 +1,85 @@
-"use client";
+'use client'
 
-import { useDraw } from "@/hooks/useDraw";
-import { useState } from "react";
-import { CompactPicker } from "react-color";
+import { FC, useEffect, useState } from 'react'
+import { useDraw } from '../hooks/useDraw'
+import { ChromePicker } from 'react-color'
 
-export default function Canvas() {
-  const [color, setColor] = useState<string>('#000');
-  const [width, setWidth] = useState<number>(5);
-  const { canvasRef, onMouseDown, clear } = useDraw(drawLine);
+import { io } from 'socket.io-client'
+import { drawLine } from '../util/drawLine'
+const socket = io('http://localhost:3001')
 
-  function drawLine({ prevPoint, currentPoint, context }: Draw) {
-    const { x: currX, y: currY } = currentPoint;
-    const lineColor = color;
-    const lineWidth = width;
+interface pageProps {}
 
-    let startPoint = prevPoint ?? currentPoint;
-    context.beginPath();
-    context.lineWidth = lineWidth;
-    context.strokeStyle = lineColor;
-    context.moveTo(startPoint.x, startPoint.y);
-    context.lineTo(currX, currY);
-    context.stroke();
+type DrawLineProps = {
+  prevPoint: Point | null
+  currentPoint: Point
+  color: string
+}
+
+const page: FC<pageProps> = () => {
+  const [color, setColor] = useState<string>('#000')
+  const { canvasRef, onMouseDown, clear } = useDraw(createLine)
+
+  useEffect(() => {
+    const ctx = canvasRef.current?.getContext('2d')
+
+    socket.emit('client-ready')
+
+    socket.on('get-canvas-state', () => {
```

Comment:
My suggestion would be to remove this event. Instead of the server telling the client to send the `canvas-state` by sending a `get-canvas-state` event. It might be better for the client to continuously send the canvas state everytime the user draws a line, so it's one less thing the server has to worry about.

Reply:
I believe we do not need this anymore